.github/workflows/frameworks.yml

```diff
@@ -0,0 +1,98 @@
+# at first, we demo HDF5 framework. More can be added.
+name: ci_frameworks
+
+on:
+  push:
+    paths:
+      - "mesonbuild/dependencies/**"
+      - ".github/workflows/frameworks.yml"
+  pull_request:
+    paths:
+      - "mesonbuild/dependencies/**"
+      - ".github/workflows/frameworks.yml"
+
+jobs:
+
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - run: python -m pip install -e .
+    - run: sudo apt install -yq --no-install-recommends ninja-build g++ gfortran libhdf5-dev
+    - run: meson setup "test cases/frameworks/25 hdf5" build
```

Comment:
All Linux dependency tests should be done on the travis image. All of these deps are already installed so this should already be run. Unless there is a good reason, this should be removed.

Reply:
I think the Linux test can be removed from Actions, since the Mac Actions test will catch the latest HDF5 release, while an older HDF5 release will be in the Linux Docker image due to the nature of non-rolling Linux releases.